packages/datadog-instrumentations/src/fs.js

```diff
@@ -0,0 +1,357 @@
+  fchmod: ['fd', 'mode'],
+  fchown: ['fd', 'uid', 'gid'],
+  fdatasync: ['fd'],
+  fstat: ['fd', 'options'],
+  fsync: ['fd'],
+  ftruncate: ['fd', 'len'],
+  futimes: ['fd', 'atime', 'mtime'],
+  lchmod: ['path', 'mode'],
+  lchown: ['path', 'uid', 'gid'],
+  link: ['existingPath', 'newPath'],
+  lstat: ['path', 'options'],
+  lutimes: ['path', 'atime', 'mtime'],
+  mkdir: ['path', 'options'],
+  mkdtemp: ['prefix', 'options'],
+  open: ['path', 'flag', 'mode'],
+  opendir: ['path', 'options'],
+  read: ['fd'],
+  readdir: ['path', 'options'],
+  readFile: ['path', 'options'],
+  readlink: ['path', 'options'],
+  readv: ['fd'],
+  realpath: ['path', 'options'],
+  rename: ['oldPath', 'newPath'],
+  rmdir: ['path', 'options'],
+  rm: ['path', 'options'],
+  stat: ['path', 'options'],
+  symlink: ['target', 'path', 'type'],
+  truncate: ['path', 'len'],
+  unlink: ['path'],
+  utimes: ['path', 'atime', 'mtime'],
+  write: ['fd'],
+  writeFile: ['file', 'data', 'options'],
+  writev: ['fd']
+}
+
+const watchMethods = {
+  unwatchFile: ['path', 'listener'],
+  watch: ['path', 'options', 'listener'],
+  watchFile: ['path', 'options', 'listener']
+}
+
+const paramsByFileHandleMethods = {
+  appendFile: ['data', 'options'],
+  chmod: ['mode'],
+  chown: ['uid', 'gid'],
+  close: [],
+  createReadStream: ['options'],
+  createWriteStream: ['options'],
+  datasync: [],
+  read: ['buffer', 'offset', 'length', 'position'],
+  readableWebStream: [],
+  readFile: ['options'],
+  readLines: ['options'],
+  readv: ['buffers', 'position'],
+  stat: ['options'],
+  sync: [],
+  truncate: ['len'],
+  utimes: ['atime', 'mtime'],
+  write: ['buffer', 'offset', 'length', 'position'],
+  writeFile: ['data', 'options'],
+  writev: ['buffers', 'position']
+}
+
+addHook({ name: 'fs' }, fs => {
+  const asyncMethods = Object.keys(paramsByMethod)
+  const syncMethods = asyncMethods.map(name => `${name}Sync`)
+
+  massWrap(fs, asyncMethods, createWrapFunction())
+  massWrap(fs, syncMethods, createWrapFunction())
+  massWrap(fs.promises, asyncMethods, createWrapFunction('promises.'))
+
+  wrap(fs.realpath, 'native', createWrapFunction('', 'realpath.native'))
+  wrap(fs.realpathSync, 'native', createWrapFunction('', 'realpath.native'))
+  wrap(fs.promises.realpath, 'native', createWrapFunction('', 'realpath.native'))
+
+  wrap(fs, 'createReadStream', createWrapCreateStream())
+  wrap(fs, 'createWriteStream', createWrapCreateStream())
+  if (fs.Dir) {
+    wrap(fs.Dir.prototype, 'close', createWrapFunction('dir.'))
+    wrap(fs.Dir.prototype, 'closeSync', createWrapFunction('dir.'))
+    wrap(fs.Dir.prototype, 'read', createWrapFunction('dir.'))
+    wrap(fs.Dir.prototype, 'readSync', createWrapFunction('dir.'))
+    wrap(fs.Dir.prototype, Symbol.asyncIterator, createWrapDirAsyncIterator())
+  }
+
+  wrap(fs, 'unwatchFile', creaetWatchWrapFunction())
+  wrap(fs, 'watch', creaetWatchWrapFunction())
+  wrap(fs, 'watchFile', creaetWatchWrapFunction())
+
+  wrapFileHandle(fs)
```

Comment:
You're calling an async function without a catch handler. If this fails for any reason it will crash the process.

Reply:
This call is not necessary anymore